### ISSUE TITLE
Newdle internal error 500

### ISSUE THREAD

**Issue by tomalin** [indico/newdle#448]
Whenever I'm asked to complete a Newdle poll (via Google Chrome  version114.0.5735.199), I see an error message such as:

"Error occurred Error: Request to /api/newdle/XXX/participants/me/busy?date=2023-07-12&tz-Europe%2FLondon failed (500) internal_error"

If I change to a different time zone, another such message appears for the new time zone. Though the times shown in the poll do change correctly.

**Comment by ThiefMaster** [comment]
I removed the newdle ID from your post, since that would allow anyone to view (and respond to) this particular newdle.

In any case I think I managed to track down the error in our logs and it fails to load free/busy information for you - most likely because you're using a non-CERN email address, and if I have to take a guess you deleted your CERN mailbox in favor of using your external email address everywhere.

This error should be caught and we should be silently returning an empty list of busy times in such cases. Just for context, this is the exception from our logs:

```
[2023-06-29 18:38:20,615] ERROR in app: Request failed
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/webargs/core.py", line 594, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/newdle/api.py", line 265, in get_participant_busy_times
    return _get_busy_times(date, tz, g.user['uid'], g.user['email'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/newdle/api.py", line 290, in _get_busy_times
    data += module.fetch_free_busy(date, tz, uid, email)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/newdle/providers/free_busy/exchange.py", line 115, in fetch_free_busy
    for busy_info in info:
  File "/usr/local/lib/python3.11/site-packages/exchangelib/services/common.py", line 225, in _elems_to_objs
    for elem in elems:
  File "/usr/local/lib/python3.11/site-packages/exchangelib/services/common.py", line 287, in _chunked_get_elements
    yield from self._get_elements(payload=payload_func(chunk, **kwargs))
  File "/usr/local/lib/python3.11/site-packages/exchangelib/services/common.py", line 308, in _get_elements
    yield from self._response_generator(payload=payload)
  File "/usr/local/lib/python3.11/site-packages/exchangelib/services/get_user_availability.py", line 47, in _get_elements_in_response
    container_or_exc = self._get_element_container(message=msg.find(f"{{{MNS}}}ResponseMessage"))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/exchangelib/services/common.py", line 606, in _get_element_container
    raise self._get_exception(code=response_code, text=msg_text, msg_xml=msg_xml)
exchangelib.errors.ErrorAddressSpaceNotFound: Microsoft.Exchange.InfoWorker.Common.Availability.AddressSpaceNotFoundException: Configuration information for forest/domain XXXX.ac.uk could not be found in Active Directory.
   at Microsoft.Exchange.InfoWorker.Common.Availability.TargetForestConfigurationCache.FindByDomain(OrganizationId organizationId, String domainName)
   at Microsoft.Exchange.InfoWorker.Common.Availability.QueryGenerator.GetTargetForestConfiguration(EmailAddress emailAddress)
. Name of the server where exception originated: GVAP278MB0761. LID: 47932
```

**Comment by tomalin** [comment]
Thanks! I understand from your reply that this is a indeed bug in Newdle, and it will be fixed at some point the future.
I do have a valid CERN Email address, but as expected from the help documentation https://account.cern.ch/account/Help/?kbid=011020 , since I work for an external institute and don't have a CERN mailbox, my primary Email address is my external institute one.

**Comment by ThiefMaster** [comment]
Yeah, if you do not have a CERN mailbox we can't get free-busy information for you. The bug is simply that you get an error in this case

**Comment by ThiefMaster** [comment]
I think I fixed this in 6ac593039 and just forgot to close this issue...
